src/main/server/serverAPI.ts

```diff
@@ -37,8 +37,12 @@ export async function getServerAPI<T>(url: URL, isAuthenticated: boolean, onSucc
             if (response.statusCode === 200) {
                 response.on('data', (chunk: Buffer) => {
                     const raw = `${chunk}`;
-                    const data = JSON.parse(raw) as T;
-                    onSuccess(data);
+                    try {
+                        const data = JSON.parse(raw) as T;
+                        onSuccess(data);
+                    } catch (e) {
+                        log.error('Error parsing server data from', url.toString());
```

Comment:
shouldn't we do a `onError?.(...)` here? so the requester know it failed?

Reply:
Seems like a good idea yeah :P